api/groups.go

```
@@ -601,3 +612,52 @@ func (api *API) GetTeamsReportLines(listOfGroups *cognitoidentityprovider.ListGr
 
 	return &GroupsUsersList, nil
 }
+
+// sortGroups sorts groups in alphabetical order based on the specified sorting criteria
+func sortGroups(w http.ResponseWriter, listGroupOutput *cognitoidentityprovider.ListGroupsOutput, sortBy []string) {
+	groups := listGroupOutput.Groups
+
+	switch {
+	case sortBy[0] == "name" && len(sortBy) == 1:
+		sortByGroupName(groups, true)
+		return
+	case sortBy[0] == "name" && len(sortBy) == 2:
+		switch sortBy[1] {
+		case "asc":
+			sortByGroupName(groups, true)
+			return
+		case "desc":
+			sortByGroupName(groups, false)
+			return
+		default:
+			http.Error(w, "incorrect sort value. Groups not sorted.", http.StatusBadRequest)
+			return
+		}
+	default:
+		http.Error(w, "incorrect sort value. Groups not sorted.", http.StatusBadRequest)
+		return
```

This is going to keep processing rather than stopping the request I think? Should return an err and handle it above.